### PR TITLE
Add Blog Post on TestStream

### PR DESCRIPTION
**Pull request by tgroh**
The post goes into details on the previous and expanded testing
infrastructure provided by the Beam SDK and provides examples of the
times at which elements arrive, and how that impacts the outputs
produced by a PipelineRunner.

### CODE REVIEW

**Comment by tgroh** [comment]
R: @francesperry

**Comment by tgroh** [comment]
I haven't yet gotten everything to render locally, fwiw

File: _posts/2016-10-17-test-stream.md [hunk 1]
```diff
@@ -0,0 +1,299 @@
+---
+layout: post
+title:  "Testing Unbounded Pipeliens in Apache Beam"
+date:   2016-10-17 10:21:48 -0800
+excerpt_separator: <!--more-->
+categories: blog beam testing java
```
Comment (francesperry) [inline _posts/2016-10-17-test-stream.md]: Don't add all these. They make the URL weird. I'd just stick to 'blog'. Also filed BEAM-776 to figure out how to make this not ugly.

File: _posts/2016-10-17-test-stream.md [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+authors:
+- tgroh
```
Comment (francesperry) [inline _posts/2016-10-17-test-stream.md]: Need to add yourself to _data/authors.yml

File: _posts/2016-10-17-test-stream.md [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+familiar with watermarks, windowing, and triggering in the Beam model,
+[Streaming 101](https://www.oreilly.com/ideas/the-world-beyond-batch-streaming-101)
+and [Streaming 102](https://www.oreilly.com/ideas/the-world-beyond-batch-streaming-102)
+are an excellent place to get started. A key takeaway from
+these articles: in realistic streaming scenarios with intermittent failures and
+disconnected users, data can arrive out of order or be delayed. Beam’s
+primitives provide a way for users to perform useful, powerful, and correct
+computations in spite of these challenges.
+
+We need comprehensive tests that cover crucial failure scenarios and corner
```
Comment (francesperry) [inline _posts/2016-10-17-test-stream.md]: Perhaps preface with "As users of Beam, we..."  to clarify this isn't referring to testing Beam itself.

File: _posts/2016-10-17-test-stream.md [hunk 4]
```diff
@@ -0,0 +1,299 @@
...
+cases to gain real confidence that a pipeline is ready for production. The
+existing testing infrastructure within the Beam SDKs permits tests to be written
+which examine the contents of a Pipeline at execution time. However, writing
+unit tests for pipelines that may receive late data or trigger multiple times
+has historically ranged from complex to not possible, as pipelines that read
+from unbounded sources do not shut down without external intervention, while
+Pipelines that read from bounded sources exclusively cannot test behavior with
```
Comment (francesperry) [inline _posts/2016-10-17-test-stream.md]: Pipelines -> pipelines?

File: _posts/2016-10-17-test-stream.md [hunk 5]
```diff
@@ -0,0 +1,299 @@
...
+late data nor most speculative triggers. Without additional tools, pipelines
+that use custom triggers and handle out-of-order data could not be easily tested.
+
+This blog post introduces our new framework for writing tests for pipelines that
+handle delayed and out-of-order data in the context of the LeaderBoard/Mobile
```
Comment (francesperry) [inline _posts/2016-10-17-test-stream.md]: Leaderboard pipeline from the Mobile Gaming example series.

File: _posts/2016-10-17-test-stream.md (hunk 1 above)
Comment (tgroh) [inline _posts/2016-10-17-test-stream.md]: Done.

(Picture me collapsing at the knees and cradling my head)

File: _posts/2016-10-17-test-stream.md (hunk 2 above)
Comment (tgroh) [inline _posts/2016-10-17-test-stream.md]: Done.

File: _posts/2016-10-17-test-stream.md (hunk 3 above)
Comment (tgroh) [inline _posts/2016-10-17-test-stream.md]: Done.

File: _posts/2016-10-17-test-stream.md (hunk 4 above)
Comment (tgroh) [inline _posts/2016-10-17-test-stream.md]: Done.

File: _posts/2016-10-17-test-stream.md (hunk 5 above)
Comment (tgroh) [inline _posts/2016-10-17-test-stream.md]: Done.

**Comment by tgroh** [comment]
Generated static content. Fixed spelling of title.

I will need to update the post date before merge, including content regeneration.

**Comment by francesperry** [comment]
Please _don't_ regenerate content/ -- just leave that whole directory out and let the person doing the final merge do it. Also, see new shiny [website instructions](http://beam.incubator.apache.org/contribute/contribution-guide/#website).

Also, you'll want to rename the file to the date you expect to publish. 10/20 seem reasonable?

**Comment by tgroh** [comment]
Gone has the content.

Moved everything over.

**Comment by francesperry** [comment]
Looks good to me. Merging.
